CMakeLists.txt

```diff
@@ -152,15 +151,31 @@ if (BUILD_FORTRAN_WRAPPER)
 
     if (Fortran_COMPILER_NAME STREQUAL "gfortran")
         # gfortran
-        set (CMAKE_Fortran_FLAGS         "-fdefault-real-8")
-        set (CMAKE_Fortran_FLAGS_RELEASE "-O3")
-        set (CMAKE_Fortran_FLAGS_DEBUG   "-g")
+        set (CMAKE_Fortran_FLAGS "-fdefault-real-8" CACHE STRING
+             "Flags used by the Fortran compiler during all builds."
+             FORCE )
+        set (CMAKE_Fortran_FLAGS_RELEASE "-O3" CACHE STRING
+             "Flags used by the Fortran compiler during Release builds."
+             FORCE )
+        set (CMAKE_Fortran_FLAGS_DEBUG "-g" CACHE STRING
+             "Flags used by the Fortran compiler during Debug builds."
+             FORCE )
     elseif (Fortran_COMPILER_NAME STREQUAL "ifort")
         # ifort (untested)
-        set (CMAKE_Fortran_FLAGS         "-r8")
-        set (CMAKE_Fortran_FLAGS_RELEASE "-O3")
-        set (CMAKE_Fortran_FLAGS_DEBUG   "-g -traceback -fpe0 -check all")
+        set (CMAKE_Fortran_FLAGS "-r8" CACHE STRING
+             "Flags used by the Fortran compiler during all builds."
+             FORCE )
+        set (CMAKE_Fortran_FLAGS_RELEASE "-O3" CACHE STRING
+             "Flags used by the Fortran compiler during Release builds."
+             FORCE )
+        set (CMAKE_Fortran_FLAGS_DEBUG "-g -traceback -fpe0 -check all" CACHE STRING
+             "Flags used by the Fortran compiler during Debug builds."
+             FORCE )
```

Comment:
Ok, thanks for your contribution. Keeping the discussion here:

    Also, it's worth pointing out that the CMAKE_CXX_FLAGS_Profile are force cached in that same file and they function as expected. I duplicated this behavior in my commit.

Ok, I think we shouldn't do that. Could you remove it also in the other places where they are cached?